pkg/util/types/types.go

```diff
@@ -73,6 +78,36 @@ func (a AWSStringSlice) Hash() *string {
 	return aws.String(output)
 }
 
+// Get the configmap that holds the whitelisted internet facing
+func GetInternetFacingConfigMap(namespace, ingressName string) map[string]string {
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		logger.Errorf(err.Error())
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		logger.Errorf(err.Error())
+	}
+	cm, err := clientset.CoreV1().ConfigMaps(namespace).Get(restrictIngressConfigMap, metav1.GetOptions{})
+	if err != nil {
+		logger.Errorf(err.Error())
+	}
+	return cm.Data
```

Comment:
I'd like to Maintain a controller-level cache of this configmap returned here.
  - I want to prevent an ingress event, which causes all ingresses to be re-evaluated, from hitting the api server N time(s) to lookup 
  - The result being, most lookups to [IngressAllowedExternal](https://github.com/coreos/alb-ingress-controller/pull/315/files#diff-902bed7fc500e15b13b452a38d1269acR101) would result in a local lookup and **not** a new `go-client` call to the k8s api server.
  - Similar to the previous comment, this logic likely best lives in the `annotations` package, in [validation.go] (https://github.com/coreos/alb-ingress-controller/blob/master/pkg/annotations/validation.go) specifically.

Reply:
The caching is addressed now here: https://github.com/coreos/alb-ingress-controller/pull/315/files#diff-f7784086caeb4e10b07629efa6b4d14aR295